import-scripts/preconsume_problematic_samples.sh

```diff
@@ -36,27 +32,32 @@ function make_tmp_dir_if_necessary() {
     fi
 }
 
+function set_cvr_fetch_url_prefix() {
+    if [ "$COHORT" == "mskimpact" ] ; then
+        CVR_FETCH_URL_PREFIX=$CVR_IMPACT_FETCH_URL_PREFIX
+    elif [ "$COHORT" == "mskimpact_heme" ] ; then
+        CVR_FETCH_URL_PREFIX=$CVR_HEME_FETCH_URL_PREFIX
+    elif [ "$COHORT" == "mskarcher" ] ; then
+        CVR_FETCH_URL_PREFIX=$CVR_ARCHER_FETCH_URL_PREFIX
+    elif [ "$COHORT" == "mskaccess" ] ; then
+        CVR_FETCH_URL_PREFIX=$CVR_ACCESS_FETCH_URL_PREFIX
+    else
+        echo "Error: Invalid cohort name '$COHORT'" >&2
+        exit 1
+    fi
+}
+
 function fetch_currently_queued_samples() {
     dmp_token=$(curl $CVR_CREATE_SESSION_URL | grep session_id | sed -E 's/",[[:space:]]*$//' | sed -E 's/.*"//')
-    curl "${CVR_IMPACT_FETCH_URL_PREFIX}/${dmp_token}/0" > ${IMPACT_FETCH_OUTPUT_FILEPATH}
-    curl "${CVR_HEME_FETCH_URL_PREFIX}/${dmp_token}/0" > ${HEME_FETCH_OUTPUT_FILEPATH}
-    curl "${CVR_ARCHER_FETCH_URL_PREFIX}/${dmp_token}/0" > ${ARCHER_FETCH_OUTPUT_FILEPATH}
-    curl "${CVR_ACCESS_FETCH_URL_PREFIX}/${dmp_token}/0" > ${ACCESS_FETCH_OUTPUT_FILEPATH}
+    curl "${CVR_FETCH_URL_PREFIX}/${dmp_token}/0" > ${FETCH_OUTPUT_FILEPATH}
 }
 
 function detect_samples_with_problematic_events() {
-    $DETECT_SAMPLES_WITH_NULL_DP_AD_FIELDS_SCRIPT_FILEPATH ${IMPACT_FETCH_OUTPUT_FILEPATH} ${IMPACT_CONSUME_IDS_FILEPATH}
-    $DETECT_SAMPLES_WITH_NULL_DP_AD_FIELDS_SCRIPT_FILEPATH ${HEME_FETCH_OUTPUT_FILEPATH} ${HEME_CONSUME_IDS_FILEPATH}
-    $DETECT_SAMPLES_WITH_NULL_DP_AD_FIELDS_SCRIPT_FILEPATH ${ACCESS_FETCH_OUTPUT_FILEPATH} ${ACCESS_CONSUME_IDS_FILEPATH}
+    $DETECT_SAMPLES_WITH_NULL_DP_AD_FIELDS_SCRIPT_FILEPATH ${FETCH_OUTPUT_FILEPATH} ${PROBLEMATIC_EVENT_CONSUME_IDS_FILEPATH}
 }
 
 function detect_samples_with_problematic_metadata() {
-    $DETECT_SAMPLES_WITH_PROBLEMATIC_METADATA_SCRIPT_FILEPATH ${ARCHER_FETCH_OUTPUT_FILEPATH} ${ARCHER_CONSUME_IDS_FILEPATH}
```

Comment:
I guess before we were only scanning archer for genepanel=UNKNOWN. This change seems to mean that we will scan all 4 cohorts for bad genepanel references (probably a good thing). Does this run smoothly though on other cohorts? (The json schema may differ)

Reply:
Yes, with the way I rewrote it, all cohorts will be checked for problematic events + problematic metadata. I figured it made sense since we already have the functionality. I didn't run into any errors testing on the 4 cohorts but the queues that I pulled didn't have any issues in them so I might need to test a bit more to make sure